### PR TITLE
Updated AzureiD in Azure plugin with providerId

### PR DESCRIPTION
**Pull request by gumbaba-gs**
## Description
Refactor - Account-layer AzureId to become ProviderId 

## Motivation and Context

#209
Contributes to hamlet-io/engine#1463

## How Has This Been Tested?

Tested with provider Id changes made on executor.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None
Closes hamlet/engine/plugin/azure/#209

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

**Review by ml019 — COMMENTED** [review]
Needs reference to base PR that will add support for ProviderId to the Account object

**Review by ml019 — DISMISSED** [review]
No issue

**Comment by ml019** [comment]
@RossMurr4y @gumbaba-gs needs rebasing before I can merge

**Comment by gumbaba-gs** [comment]
> @RossMurr4y @gumbaba-gs needs rebasing before I can merge

@ml019  @RossMurr4y re-based and resolved conflicts.     f08a9f0

File: azure/deploymentframeworks/arm/output.ftl [hunk 1]
```diff
@@ -137,7 +137,7 @@
 [#function pseudoArmStackOutputScript description outputs filesuffix=""]
     [#local outputString = ""]
 
-    [#list getArmTemplateCoreOutputs(region, accountObject.AzureId, commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
+    [#list getArmTemplateCoreOutputs(region, accountObject.ProviderId, commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
```
Comment (RossMurr4y) [inline azure/deploymentframeworks/arm/output.ftl:140]:
```suggestion
    [#list getArmTemplateCoreOutputs(region, accountObject.ProviderId!{}, commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
```

File: azure/deploymentframeworks/arm/output.ftl [hunk 2]
```diff
@@ -281,7 +281,8 @@
     [#--          The marker indicates the minimum scope to which a resource exists.                --]
     [#local resourceProfileScope = getAzureResourceProfile(getResourceType(id)).scope]
     [#local currentScope = {
-        "Subscription" : accountObject.AzureId,
+        "Subscription" : accountObject.ProviderId,
```
Comment (RossMurr4y) [inline azure/deploymentframeworks/arm/output.ftl:284]:
```suggestion
        "Subscription" : accountObject.ProviderId!{},
```

File: azure/deploymentframeworks/arm/output.ftl [hunk 3]
```diff
@@ -281,7 +281,8 @@
     [#--          The marker indicates the minimum scope to which a resource exists.                --]
     [#local resourceProfileScope = getAzureResourceProfile(getResourceType(id)).scope]
     [#local currentScope = {
-        "Subscription" : accountObject.AzureId,
+        "Subscription" : accountObject.ProviderId!{},
```
Comment (RossMurr4y) [inline azure/deploymentframeworks/arm/output.ftl:284]:
```suggestion
        "Subscription" : accountObject.ProviderId!"",
```

File: azure/deploymentframeworks/arm/output.ftl [hunk 4]
```diff
@@ -137,7 +137,7 @@
 [#function pseudoArmStackOutputScript description outputs filesuffix=""]
     [#local outputString = ""]
 
-    [#list getArmTemplateCoreOutputs(region, accountObject.AzureId, commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
+    [#list getArmTemplateCoreOutputs(region, accountObject.ProviderId!{}, commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
```
Comment (RossMurr4y) [inline azure/deploymentframeworks/arm/output.ftl:140]:
```suggestion
    [#list getArmTemplateCoreOutputs(region, accountObject.ProviderId!"", commandLineOptions.Deployment.ResourceGroup.Name) as key,value]
```

**Review by RossMurr4y — APPROVED** [review]

**Pull request merged**